eigen/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+# features. Defining _GNU_SOURCE causes Eigen to try to use them, which fails.
+# Use _DEFAULT_SOURCE instead, it provides enough features to build Eigen.
+set(extra_defines -D_DEFAULT_SOURCE)
+# Final flags for C and C++:
+set(c_flags "${includes} ${extra_defines} -D_DEFAULT_SOURCE=1")
+set(cxx_flags "${includes} ${extra_defines} -D_DEFAULT_SOURCE=1")
+
+set(assert_flags "${assert_flags} -DDEBUG -D_DEBUG -ffunction-sections -fdata-sections ")
+
+if(CONFIG_COMPILER_OPTIMIZATION_DEFAULT)
+    set(opt_flags "-Og -ggdb -mlongcalls ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug
+                 -DCMAKE_C_FLAGS_DEBUG=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_DEBUG=${opt_flags})
+elseif(CONFIG_COMPILER_OPTIMIZATION_SIZE)
+    # Normally we should use MinSizeRel build type for this,
+    # however Eigen CMake files only handle Debug and Release
+    # (see e.g. Eigen/cmake/EigenCompilerOptions.cmake).
+    # So we redefine the flags for Release instead.
+    set(opt_flags "-Os -ggdb -mlongcalls ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Release
+                 -DCMAKE_C_FLAGS_RELEASE=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_RELEASE=${opt_flags})
+elseif(COMPILER_OPTIMIZATION_PERF)
+    # Currently Eigen fails to compile at -O2 level on Xtensa due to a compiler issue,
+    # details in https://github.com/jcmvbkbc/gcc-xtensa/issues/14.
+    # Compiling at -O3 happens to be okay.
+    set(opt_flags "-O3 -ggdb -mlongcalls ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Release
+                 -DCMAKE_C_FLAGS_RELEASE=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_RELEASE=${opt_flags})
+elseif(COMPILER_OPTIMIZATION_NONE)
+    set(opt_flags "-O0 -ggdb -mlongcalls ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug
+                 -DCMAKE_C_FLAGS_DEBUG=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_DEBUG=${opt_flags})
+else()
+    message(FATAL_ERROR "Unsupported optimization level")
+endif()
+
+# Python to use for Eigen build
+idf_build_get_property(python PYTHON)
+
+include(ExternalProject)
+
+# Build Eigen in this directory:
+set(BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/eigen-build)
+
+# Add Eigen as a subproject.
+ExternalProject_Add(eigen_proj
+    SOURCE_DIR ${COMPONENT_DIR}/eigen
+    BINARY_DIR ${BINARY_DIR}
+    BUILD_BYPRODUCTS ${all_libraries}
+    # These two options are set so that Ninja immediately outputs
+    # the subproject build to the terminal. Otherwise it looks like the
+    # build process "hangs" for too long until Eigen build is complete.
+    USES_TERMINAL_CONFIGURE TRUE
+    USES_TERMINAL_BUILD TRUE
+    # Arguments to pass to Eigen CMake invocation:
+    CMAKE_ARGS
+        -DCMAKE_C_FLAGS=${c_flags}
+        -DCMAKE_CXX_FLAGS=${cxx_flags}
+        ${opt_args}
+        -DCMAKE_INSTALL_PREFIX=${BINARY_DIR}/install
+        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+        -DCMAKE_SYSTEM_PROCESSOR=${CONFIG_IDF_TARGET_ARCH}
```

Comment:
I haven't found any references in `CMAKE_SYSTEM_PROCESSOR` in libeigen. Is this line necessary?